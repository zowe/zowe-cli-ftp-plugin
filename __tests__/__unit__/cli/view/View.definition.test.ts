/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import * as ViewDefinition from "../../../../src/cli/view/View.definition";

describe("View definition", () => {

    it("should match the snapshot", () => {
        ViewDefinition.children.forEach((child) => {
            child.handler = child.handler.replace(/^.*[\\/]/, "");
        });
        expect(ViewDefinition).toMatchSnapshot();
    });

});
