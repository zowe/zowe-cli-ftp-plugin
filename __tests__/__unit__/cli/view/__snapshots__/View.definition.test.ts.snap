// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View definition should match the snapshot 1`] = `
Object {
  "aliases": Array [
    "vw",
  ],
  "children": Array [
    Object {
      "aliases": Array [
        "ds",
      ],
      "description": "View the contents of a z/OS data set",
      "examples": Array [
        Object {
          "description": "View the content of the data set \\"ibmuser.cntl(iefbr14)\\"",
          "options": "\\"ibmuser.cntl(iefbr14)\\"",
        },
        Object {
          "description": "View the content of the data set \\"ibmuser.loadlib(main)\\" and pipe it into the hex viewer program xxd",
          "options": "\\"ibmuser.loadlib(main)\\" -b | xxd",
        },
      ],
      "handler": "DataSet.Handler",
      "name": "data-set",
      "options": Array [
        Object {
          "aliases": Array [
            "b",
          ],
          "description": "View content in binary form without converting to ASCII text",
          "name": "binary",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "ec",
          ],
          "defaultValue": null,
          "description": "The encoding for download and upload of z/OS data set.",
          "name": "encoding",
          "type": "string",
        },
      ],
      "positionals": Array [
        Object {
          "description": "The data set (PDS member or physical sequential data set) which you would like to view the contents of.",
          "name": "dataSet",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "View  data set content",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "sfbi",
      ],
      "description": "View the contents of a spool file from a z/OS job on spool/JES queues. The command does not pre-validate the JOBID or spool ID.",
      "examples": Array [
        Object {
          "description": "View the spool file with ID 4 for the job with job ID JOB00123",
          "options": "JOB00123 4",
        },
      ],
      "handler": "SpoolFileById.Handler",
      "name": "spool-file-by-id",
      "positionals": Array [
        Object {
          "description": "The z/OS JOBID of the job containing the spool file you want to view. No pre-validation of the JOBID is performed.",
          "name": "jobid",
          "required": true,
          "type": "string",
        },
        Object {
          "description": "The spool file ID number for the spool file to view. No pre-validation of the ID is performed.",
          "name": "spoolfileid",
          "required": true,
          "type": "number",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "View a spool file from a z/OS job",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "jsbj",
      ],
      "description": "View status details of a single z/OS job on spool/JES queues. The command does not prevalidate the JOBID.",
      "examples": Array [
        Object {
          "description": "View the status for the job with ID \\"JOB00123\\" (optionally abbreviating the ID)",
          "options": "j123",
        },
      ],
      "handler": "JobStatusByJobId.Handler",
      "name": "job-status-by-jobid",
      "options": Array [],
      "outputFormatOptions": true,
      "positionals": Array [
        Object {
          "description": "The ID of the jobfor which you would like to list spool files",
          "name": "jobid",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "View status details of a z/OS job",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "uss",
        "uf",
      ],
      "description": "View the contents of a Unix System Services File",
      "examples": Array [
        Object {
          "description": "View the content of the USS file \\"/u/users/ibmuser/myfile.txt\\"",
          "options": "\\"/u/users/ibmuser/myfile.txt\\"",
        },
        Object {
          "description": "View the content of the USS file \\"/u/users/ibmuser/myjava.jar\\" in binary mode and pipe it into the hex viewer command xxd",
          "options": "\\"/u/users/ibmuser/myjava.jar\\" -b | xxd ",
        },
      ],
      "handler": "UssFile.Handler",
      "name": "uss-file",
      "options": Array [
        Object {
          "aliases": Array [
            "b",
          ],
          "description": "View content in binary form without converting to ASCII text",
          "name": "binary",
          "type": "boolean",
        },
      ],
      "positionals": Array [
        Object {
          "description": "The USS file you'd like to view the contents of.",
          "name": "ussFile",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "View  USS file content",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "asbj",
      ],
      "description": "View all spool content for a job by providing the job id",
      "examples": Array [
        Object {
          "description": "View all spool content for the job with ID JOB00123 (optionally abbreviating the job ID)",
          "options": "j123",
        },
      ],
      "handler": "AllSpoolByJobId.Handler",
      "name": "all-spool-by-jobid",
      "options": Array [],
      "positionals": Array [
        Object {
          "description": "The ID of the job for which you would like to list spool files",
          "name": "jobid",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "View all spool content for a job",
      "type": "command",
    },
  ],
  "description": "View data sets, job output, and USS content",
  "name": "view",
  "passOn": Array [
    Object {
      "ignoreNodes": Array [
        Object {
          "type": "group",
        },
      ],
      "merge": true,
      "property": "options",
      "value": Array [
        Object {
          "aliases": Array [
            "H",
          ],
          "description": "The hostname or IP address of the z/OS server to connect to.",
          "group": "FTP Connection options",
          "name": "host",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "P",
          ],
          "defaultValue": 21,
          "description": "The port of the z/OS FTP server.",
          "group": "FTP Connection options",
          "name": "port",
          "type": "number",
        },
        Object {
          "aliases": Array [
            "u",
          ],
          "description": "Username for authentication on z/OS",
          "group": "FTP Connection options",
          "name": "user",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "p",
            "pass",
            "pw",
          ],
          "description": "Password to authenticate to FTP.",
          "group": "FTP Connection options",
          "name": "password",
          "type": "string",
        },
        Object {
          "defaultValue": true,
          "description": "Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990). Note: Unfortunately, this plugin's functionality only works with FTP and FTPS, not 'SFTP' which is FTP over SSH.",
          "group": "FTP Connection options",
          "name": "secure-ftp",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "ru",
          ],
          "defaultValue": null,
          "description": "Reject self-signed certificates. Only specify this if you are connecting to a secure FTP instance.",
          "group": "TLS / Secure Connection options",
          "name": "reject-unauthorized",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "sn",
          ],
          "defaultValue": null,
          "description": "Server name for the SNI (Server Name Indication) TLS extension. Only specify if you are connecting securely",
          "group": "TLS / Secure Connection options",
          "name": "server-name",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "ct",
          ],
          "defaultValue": 10000,
          "description": "How long (in milliseconds) to wait for the control connection to be established.",
          "group": "FTP Connection options",
          "name": "connection-timeout",
          "type": "number",
        },
      ],
    },
  ],
  "summary": "View data set, job output, and USS content",
  "type": "group",
}
`;
